ci-operator/step-registry/ibmcloud/provision/byok/ibmcloud-provision-byok-commands.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output   
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..." 
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+    ibmcloud plugin list
+}
+
+
+function create_resource_group() {
+    local rg="$1"
+    echo "create resource group ... ${rg}"
+    "${IBMCLOUD_CLI}" resource group-create ${rg} || return 1
+    "${IBMCLOUD_CLI}" target -g ${rg} || return 1
+}
+
+ibmcloud_login
+
+## Create the instances for BYOK
+echo "$(date -u --rfc-3339=seconds) - Creating the instance for BYOK..."
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+resource_group="${CLUSTER_NAME}-rg"
+
+create_resource_group ${resource_group}
+
+instance_name="${CLUSTER_NAME}-kp"
+
+ibmcloud resource service-instance-create ${instance_name} kms tiered-pricing ${region}
+
+id=$(ibmcloud resource service-instance ${instance_name} --id -q | awk '{print $2}')
+
+export KP_INSTANCE_ID=${id}
+
+key_meterial=$(openssl rand -base64 32)
+key_name="${instance_name}-key"
+ibmcloud kp key create ${key_name} -k ${key_meterial}
+
+keyID=$(ibmcloud kp keys -l ${key_name} -c -o JSON | jq -r .[].crn)
+
+cat > "${SHARED_DIR}/ibm_kpKey.yaml" << EOF
+platform:
+  ibmcloud:
+    defaultMachinePlatform:
+      bootVolume:
+        encryptionKey: "${keyID}"
```

Comment:
Here we have to use key CRN, can we use key ID?

Reply:
If the install-config.yaml only respect crn, then suggest to save the key id in `ibm_kp.json` to avoid bother install-config.yaml when deprovision.

Reply:
1. tried with id instead with crn failed [job](https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Flexy-install/253433/console)

> 12-20 16:29:44.152  [INFO] Generating manifests files.....
12-20 16:29:44.153  level=error msg=failed to fetch Master Machines: failed to load asset "Install Config": failed to create install config: invalid "install-config.yaml" file: platform.ibmcloud.defaultMachinePlatform.bootVolume.encryptionKey: Invalid value: "f6e07c35-8494-430f-af6e-2497f509a5ff": encryptionKey is not a valid IBM CRN

2. In order to find out the kp key, create the cluster in the resource group of the kp instance and kp key, use pre defined resouce group in install-config.yaml.

Reply:
how about create 3 files:
ibmcloud_resource_group save resource_group
ibm_kp save instance id
ibm_kp_key save key_crn

Reply:
one json for deprovision, one yaml for install-config.yaml